### ISSUE TITLE
docker 1.2.x with LXC 1.0.7 and Redhad6.x gives error - Docker container factory registration failed: docker found, but not using native exec driver

### ISSUE THREAD

**Issue by babusatasiya** [google/cadvisor#873]
I am working on testing cAdviser and it is giving following error message, We are using Redhat 6.x with Lxe 1.0.7, docker 1.2.x

docker command

ocker run  --privileged=true  --volume=/:/rootfs:ro   --volume=/var/run:/run:rw   --volume=/sys:/sys:ro   --volume=/var/lib/docker/:/var/lib/docker:ro   --volume=/cgroup/cpu:/cgroup/cpu   --volume=/cgroup/cpuacct:/cgroup/cpuacct   --volume=/cgroup/cpuset:/cgroup/cpuset   --volume=/cgroup/memory:/cgroup/memory   --volume=/cgroup/blkio:/cgroup/blkio --privileged=true  --publish=8080:8080   --detach=true   --name=cadvisor   google/cadvisor:0.14.0 --logtostderr --v=5 --log_cadvisor_usage=true

Docker container factory registration failed: docker found, but not using native exec drive
I0830 02:22:35.208434 00001 storagedriver.go:100] No backend storage selected
I0830 02:22:35.208485 00001 storagedriver.go:102] Caching stats in memory for 2m0s
I0830 02:22:35.208582 00001 manager.go:123] cAdvisor running in container: "/lxc/d948848fed19676e6a18367483d03a81d0ea51a8be0f381f9285c76b0223e27f"
I0830 02:22:35.209685 00001 fs.go:88] Filesystem partitions: map[/dev/mapper/docker-252:2-395482-d948848fed19676e6a18367483d03a81d0ea51a8be0f381f9285c76b0223e27f:{mountpoint:/ major:253 minor:3} /dev/vda2:{mountpoint:/etc/resolv.conf major:252 minor:2} /dev/mapper/docker-252:2-267511-b90456ee57ed1c00fb495a4410f2b68c24c96fcc2c15537475d05901b1b9b7bf:{mountpoint:/rootfs/var/lib/docker.old/devicemapper/mnt/b90456ee57ed1c00fb495a4410f2b68c24c96fcc2c15537475d05901b1b9b7bf major:253 minor:2}]
I0830 02:22:35.212441 00001 machine.go:224] Couldn't collect info from any of the files in "/etc/machine-id,/var/lib/dbus/machine-id"
I0830 02:22:35.212535 00001 manager.go:144] Machine: {NumCores:2 CpuFrequency:2000022 MemoryCapacity:4016840704 MachineID: SystemUUID:F61A57F3-4ED5-4FF2-88AA-5FF922B9F660 BootID:8299751a-d442-4878-9060-7ac587ca9d7d Filesystems:[{Device:/dev/mapper/docker-252:2-395482-d948848fed19676e6a18367483d03a81d0ea51a8be0f381f9285c76b0223e27f Capacity:10434699264} {Device:/dev/vda2 Capacity:20474130432} {Device:/dev/mapper/docker-252:2-267511-b90456ee57ed1c00fb495a4410f2b68c24c96fcc2c15537475d05901b1b9b7bf Capacity:10434699264}] DiskMap:map[253:0:{Name:dm-0 Major:253 Minor:0 Size:107374182400 Scheduler:none} 253:1:{Name:dm-1 Major:253 Minor:1 Size:107374182400 Scheduler:none} 253:2:{Name:dm-2 Major:253 Minor:2 Size:10737418240 Scheduler:none} 253:3:{Name:dm-3 Major:253 Minor:3 Size:10737418240 Scheduler:none} 252:0:{Name:vda Major:252 Minor:0 Size:42949672960 Scheduler:cfq}] NetworkDevices:[{Name:eth0 MacAddress:fa:16:3e:a6:03:47 Speed:0 Mtu:1500}] Topology:[{Id:0 Memory:4294545408 Cores:[{Id:0 Threads:[0] Caches:[{Size:32768 Type:Data Level:1} {Size:32768 Type:Instruction Level:1} {Size:4194304 Type:Unified Level:2}]}] Caches:[]} {Id:1 Memory:0 Cores:[{Id:0 Threads:[1] Caches:[{Size:32768 Type:Data Level:1} {Size:32768 Type:Instruction Level:1} {Size:4194304 Type:Unified Level:2}]}] Caches:[]}]}
I0830 02:22:35.213669 00001 manager.go:151] Version: {KernelVersion:2.6.32-504.3.3.el6.x86_64 ContainerOsVersion:Buildroot 2014.02 DockerVersion:1.2.0 CadvisorVersion:0.14.0}
E0830 02:22:35.320652 00001 manager.go:186] Docker container factory registration failed: docker found, but not using native exec driver.
I0830 02:22:35.323261 00001 factory.go:64] Registering Raw factory
I0830 02:22:35.613342 00001 manager.go:930] Started watching for new ooms in manager
E0830 02:22:35.614256 00001 manager.go:217] Failed to start OOM watcher, will not get OOM events: exec: "journalctl": executable file not found in $PATH
I0830 02:22:35.614316 00001 factory.go:75] Using factory "raw" for container "/"
I0830 02:22:35.614595 00001 manager.go:727] Added container: "/" (aliases: [], namespace: "")
I0830 02:22:35.614790 00001 handler.go:322] Added event &{/ 0001-01-01 00:00:00 +0000 UTC containerCreation {<nil>}}
I0830 02:22:35.614888 00001 container.go:291] Start housekeeping for container "/"
I0830 02:22:35.616280 00001 manager.go:230] Starting recovery of all containers
I0830 02:22:35.618357 00001 factory.go:75] Using factory "raw" for container "/docker"
I0830 02:22:35.618567 00001 manager.go:727] Added container: "/docker" (aliases: [], namespace: "")
I0830 02:22:35.618713 00001 handler.go:322] Added event &{/docker 0001-01-01 00:00:00 +0000 UTC containerCreation {<nil>}}
I0830 02:22:35.618773 00001 container.go:291] Start housekeeping for container "/docker"
I0830 02:22:35.619119 00001 factory.go:75] Using factory "raw" for container "/lxc"
I0830 02:22:35.619303 00001 manager.go:727] Added container: "/lxc" (aliases: [], namespace: "")
I0830 02:22:35.619502 00001 handler.go:322] Added event &{/lxc 0001-01-01 00:00:00 +0000 UTC containerCreation {<nil>}}
I0830 02:22:35.619546 00001 container.go:291] Start housekeeping for container "/lxc"
I0830 02:22:35.620738 00001 factory.go:75] Using factory "raw" for container "/lxc/d948848fed19676e6a18367483d03a81d0ea51a8be0f381f9285c76b0223e27f"
I0830 02:22:35.620953 00001 manager.go:727] Added container: "/lxc/d948848fed19676e6a18367483d03a81d0ea51a8be0f381f9285c76b0223e27f" (aliases: [], namespace: "")
I0830 02:22:35.621123 00001 handler.go:322] Added event &{/lxc/d948848fed19676e6a18367483d03a81d0ea51a8be0f381f9285c76b0223e27f 0001-01-01 00:00:00 +0000 UTC containerCreation {<nil>}}
I0830 02:22:35.621186 00001 manager.go:235] Recovery completed
I0830 02:22:35.623329 00001 container.go:291] Start housekeeping for container "/lxc/d948848fed19676e6a18367483d03a81d0ea51a8be0f381f9285c76b0223e27f"
I0830 02:22:35.624473 00001 cadvisor.go:90] Starting cAdvisor version: "0.14.0" on port 8080

**Comment by rjnagal** [comment]
This is a warning that we will not be able to report some of the stats that we only get from docker using the libcontainer exec driver. You should still be able to see regular cpu, memory stats, but they show under "subcontainers" rather than docker containers (eg. /lxc container in the logs above).

You can ignore these errors if the stats you are interested in are still showing up.

**Comment by banusatasiya** [comment]
Yes, I manage to make it work, We use docker with option bridge=none and
 setup our own bridge and specify part of lxc, once I remove this option it
works as you explained.

If you got any insights why we could not work would be gr8.

**Comment by dims** [comment]
Closing this issue as it has been open since before January 1, 2024. If this is still relevant, please feel free to reopen or create a new issue.
